src/scikit_build_core/_compat/importlib/readers.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+if sys.version_info < (3, 10):
+    # Readers and MultiplexedPath were introduced in 3.10, so nothing should output a MultiplexedPath
+    # It is also tricky because it is unclear if the `resource_loader` when calling `import` would create
+    # either importlib.readers.MultiplexedPath or importlib_resources.MultiplexedPath.
+    # Creating a dummy class instead so that if it fails, it fails completely (and mypy is made happy)
+    class MultiplexedPath:
+        _paths: list[Path]
+else:
+    # From 3.11 this is an alias of importlib.resources.readers
+    from importlib.readers import MultiplexedPath
```

Comment:
This will only fix 3.11+, as it's not checking all three possibilities. We should probably check all three possibilities instead.

Reply:
Which 3 possibilities? `importlib.readers`, `importlib.resources.readers` and the dummy class? I didn't add the distinction between the first 2 because it is forwarded right now and it has a nicer parallel with the `_compat` naming scheme here :shrug:

Reply:
What about `importlib_resources`?

Reply:
> What about `importlib_resources`?

Yeah that's one that really confused me, would it be able to encounter `importlib_resources.reader.MultiplexedPath`. Seems unlikely to me, but we have a catch-all for if it doesn't recognize the type, and it might be interesting to first learn of the situation where that occurs.

Oh you've removed that catch-all though.

Reply:
Here's a proposed version that just looks for `_paths`. If a class has it, it uses it. This is safer if they move/remove `_paths` in the future, you should get a better error/failure. This should mostly be Path or MultiplexedPath, so I think it's acceptable here to just check for the attribute (duck typing). We could request a public attribute be provided in the future.

Reply:
Hmm, guess it would work. I think it would be nice to still report the unexpected situation and maybe debug the types because this part fails silently later on, e.g. for `MultiplexedPath` it just showed its `repr` and that might happen for the unexpected cases as well.

Reply:
Btw: https://github.com/python/importlib_resources/pull/310. Probably will be a fun time catching the version for that interface addition.

Reply:
We could try `paths` first.

If this fails, it shouldn't affect the error message, so I don't think it needs special handling.

Reply:
Which error message? In the case of the original this failed silently i.e. no error message in either scikit-build-core or cmake. It just failed to have any effect.

Reply:
This is technically a Traversable, and Traversable doesn't support conversion to str like Path does. I think we need to ensure we use `__fspath__` instead of `__str__`.